pride-hg-support/src/main/java/com/prezi/pride/vcs/hg/HgVcsSupport.java

```diff
@@ -0,0 +1,210 @@
+		getStatusInternal(targetDirectory, builder);
+		VcsStatus status = builder.build();
+		return status.hasUncommittedChanges() || status.hasUnpublishedChanges();
+	}
+
+	@Override
+	public VcsStatus getStatus(File targetDirectory) throws IOException {
+		final VcsStatus.Builder status = VcsStatus.builder(getRevision(targetDirectory));
+		status.withBranch(getBranch(targetDirectory));
+		getStatusInternal(targetDirectory, status);
+		return status.build();
+	}
+
+	private void getStatusInternal(File targetDirectory, final VcsStatus.Builder status) throws IOException {
+		checkModified(targetDirectory, status);
+		checkOutgoing(targetDirectory, status);
+	}
+
+	private void checkModified(File targetDirectory, final VcsStatus.Builder status) throws IOException {
+		Process process = ProcessUtils.executeIn(targetDirectory, Arrays.asList("hg", "status"), false, false);
+		String result = new String(ByteStreams.toByteArray(process.getInputStream()), Charsets.UTF_8);
+		status.withUncommittedChanges(!result.isEmpty());
+	}
+
+	private void checkOutgoing(File targetDirectory, final VcsStatus.Builder status) throws IOException {
+		boolean unpublishedChanges = hasUnpublishedChanges(targetDirectory);
+		status.withUnpublishedChanges(unpublishedChanges);
+	}
+
+	private boolean hasUnpublishedChanges(File targetDirectory) throws IOException {
+		try {
+			Process process = ProcessUtils.executeIn(targetDirectory, Arrays.asList("hg", "outgoing", "--quiet", "--template", "{node}\\n"), false, false);
+			String result = new String(ByteStreams.toByteArray(process.getInputStream()), Charsets.UTF_8);
+			return !result.isEmpty();
+		} catch (PrideException e) {
+			// `hg outgoing` exits (stupidly) with status code 1 when there are no outgoing changes
+			return false;
+		}
+	}
+
+	private String getRevision(File targetDirectory) throws IOException {
+		Process process = ProcessUtils.executeIn(targetDirectory, Arrays.asList("hg", "id", "--id"), false, false);
+		String result = new String(ByteStreams.toByteArray(process.getInputStream()), Charsets.UTF_8);
+		return result.trim().replace("+", "");
+	}
+
+	@Override
+	public String getBranch(File targetDirectory) throws IOException {
+		Process process = ProcessUtils.executeIn(targetDirectory, Arrays.asList("hg", "branch"), false, false);
+		String result = new String(ByteStreams.toByteArray(process.getInputStream()), Charsets.UTF_8);
+		return result.trim();
+	}
+
+	@Override
+	public String getDefaultBranch() throws IOException {
+		return "default";
+	}
+
+	@Override
+	public void activate(String repositoryUrl, File targetDirectory) throws IOException {
+		log.debug("Activating '{}' with url '{}'", targetDirectory, repositoryUrl);
+		File hgrcFile = new File(targetDirectory, ".hg/hgrc");
+		String hgrcText = FileUtils.readFileToString(hgrcFile, Charsets.UTF_8);
+		String activated = hgrcText.replaceAll("default = .*", "default = " + repositoryUrl);
+		FileUtils.write(hgrcFile, activated, Charsets.UTF_8);
```

Comment:
Can you explain what's happening here? Why do we need to manually edit a config file? Can we do this via some Mercurial command instead? Say, `hg config`?